### PR TITLE
Fix MaxFreeAllocator.allocateBlock

### PR DESCRIPTION
**Pull request by secfree**
### What changes are proposed in this pull request?

Fix the return value of `MaxFreeAllocator.allocateBlock`.

### Why are the changes needed?

Fix https://github.com/Alluxio/alluxio/issues/16106

### Does this PR introduce any user facing changes?

No

### CODE REVIEW

**Review by yyongycy — APPROVED** [review]
This change LGTM, 

@beinan  Please help double check.

File: core/server/worker/src/test/java/alluxio/worker/block/allocator/MaxFreeAllocatorTest.java [hunk 1]
```diff
@@ -103,7 +103,7 @@ public void allocateBlock() throws Exception {
     //  1               ├─── 3000
     //  2               └─── 3000
```
Comment (dbw9580) [inline core/server/worker/src/test/java/alluxio/worker/block/allocator/MaxFreeAllocatorTest.java:104]: Can you please also update the diagram in the comments to reflect the correct allocation results after this bug is fixed?

**Review by dbw9580 — APPROVED** [review]
LGTM, thanks for fixing this!

File: core/server/worker/src/test/java/alluxio/worker/block/allocator/MaxFreeAllocatorTest.java (hunk 1 above)
Comment (secfree) [reply core/server/worker/src/test/java/alluxio/worker/block/allocator/MaxFreeAllocatorTest.java:104]: Fixed, please help check.

**Review by dbw9580 — APPROVED** [review]

**Comment by HelloHorizon** [comment]
alluxio-bot, merge this please

**Pull request merged**